### PR TITLE
Create command to update atomist dependencies

### PR DESCRIPTION
**Pull request by lievendoclo**

### CODE REVIEW

File: lib/updateSdm.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+/**
+ * Search for an SDM extension pack.
+ *
+ * @param opts see InstallOptions
+ * @return integer return value
+ */
+export async function updateSdm(opts: InstallOptions): Promise<number> {
+    if (hasPackageJson(opts.cwd)) {
+        const spinner = createSpinner(
+            `Updating dependencies in package.json`);
```
Comment (ddgenome) [inline lib/updateSdm.ts]: Why is this broken across two lines?

File: lib/updateSdm.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        const packageJson = await fs.readJson(path.join(opts.cwd, "package.json"));
```
Comment (ddgenome) [inline lib/updateSdm.ts]: Seems a try/catch is in order.

File: lib/updateSdm.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        const extractedDependencies = getAtomistDependencies(packageJson.dependencies);
+        const extractedDevDependencies = getAtomistDependencies(packageJson.devDependencies);
+        await updateDependenciesToTag(extractedDependencies, opts.versionTag, spinner);
```
Comment (ddgenome) [inline lib/updateSdm.ts]: Should we not check if such a tagged version exists? Perhaps some of the code used in the command handler could be used here?

File: index.ts [hunk 4]
```diff
@@ -164,6 +165,22 @@ function setupYargs(yargBuilder: yb.YargBuilder): void {
             registry: argv.registry,
         })),
     });
+    yargBuilder.withSubcommand({
+        command: "update sdm",
```
Comment (ddgenome) [inline index.ts]: This command is updating more than just "sdm". I'd really like to start to get a handle on the CLI commands. They are current all over the place. But I guess that is for v2.

**Comment by cdupuis** [comment]
@lievendoclo what is the state of this?

File: lib/updateSdm.ts (hunk 2 above)
Comment (lievendoclo) [reply lib/updateSdm.ts]: Done.

File: lib/updateSdm.ts (hunk 1 above)
Comment (lievendoclo) [reply lib/updateSdm.ts]: Copy paste from other location. Fixed.

**Comment by cdupuis** [comment]
Pull request auto merged by Atomist.

* 1 approved review by @ddgenome
* 2 successful checks

[atomist:generated] [auto-merge:on-approve]

**Pull request merged**